docs/dev/app/listings.md

```diff
@@ -1,9 +1,388 @@
+The easiest way to create a new profile match field is to create a new constant of type `MatchField` as defined below. 
+```ts
+interface MatchField<T extends Listing> {
+  getMatch: (data: T, profileData: Profile) => boolean | undefined;
+  getText: (
+    data: T,
+    profileData: Profile,
+    match: boolean | undefined,
+  ) => string;
+}
+```
+- `getMatch`: takes in the listing `data` and `profileData`, and returns whether the user's information match's the listing's requirements. Returns undefined if current listing does not have information for the current field. The returned value determines the matchIcon (handled in `renderIconGroup`) and the value of getText.
+- `getText`: takes in the listing `data`, `profiledata`, and `match` (from `getMatch` above). Returns a string of what should be displayed next to the icon. 
+
+Then, call `renderIconGroup` on the match field, as below: 
+```tsx
+const newFieldMatch: MatchField<NewListingType> = {
+    getMatch: (data, profileData) => some_boolean_or_undefined_value
+    getText: (data, profileData, match) => some_text
+};
+
+export default function ProfileMatch(...) {
+    return (
+        <Flex $direction="column" $gap="16px" $h="100%" $maxW="40%">
+            <H3>Profile Match</H3>
+            {renderIconGroup(newFieldMatch as MatchField<Listing>)}
+            {/* other match fields */}
+        </Flex>
+}
+```
+
+### `InterestForm`
+- For a logged in and onboarded user, `InterestForm` enables a user to submit an interest for the current listing. This interest is saved in the `interests` table. 
+- `handleInsert` is used to handle the insertion of the user's responses to the interest form into the `interests table`. After error-checking that all required fields (according to listing type) are valid and filled out, the user's responses are upserted. If the a user submits multiple forms for one case, only the most recent response will be saved on the `interests` table.
+
+**InterestForm Props**
+- `listingData`: *Listing*. The listing whose InterestForm will be rendered. The rendering of fields is dependent on the `listing_type` of `listingData`. 
+- (optional) `interpretation`: *boolean*. Default value: false. Should be `true` for Langauge Support listings. Used to distinguish between Case Interpretations and Cases. 
+
+**Example Usage**
+- Langauge Support listings: `<InterestForm listingData={listingData} interpretation />`
+- Non-Langauge Support listings: `<InterestForm listingData={listingData} />`
+
+#### How to Customize InterestForm Questions 
+**Creating a New Question**
+1. Create a new Input Component and a new state to track the input. Update useEffect to reset the state when listingData changes. 
+2. Add a new field to the `Responses` type, corresponding to this new input 
+3. In `handleInsert`, add an error check if this new field is required for certain listing types 
+4. In `handleInsert`, conditionally add the new input to `responses` to be inserted into supabase
+
+```tsx
```

Comment:
instead of this long chunk of code, it's better to ask the reader to reference the existing codebase and only include short, relevant code blocks - this makes it a lot easier for the reader to understand what to focus on and what to change, one step at a time

if everything must be noted, it's better to split it into separate chunks rather than a big block like this

Reply:
gotcha, i split up the code chunk